pydantic/main.py

```diff
@@ -810,6 +564,383 @@ def model_parametrized_name(cls, params: tuple[type[Any], ...]) -> str:
+        exclude: AbstractSetIntStr | MappingIntStrAny | None = None,
+        update: typing.Dict[str, Any] | None = None,  # noqa UP006
+        deep: bool = False,
+    ) -> Model:
+        """
+        This method is now deprecated; use `model_copy` instead. If you need include / exclude, use:
+
+            data = self.model_dump(include=include, exclude=exclude, round_trip=True)
+            data = {**data, **(update or {})}
+            copied = self.model_validate(data)
+        """
+        warnings.warn(
+            'The `copy` method is deprecated; use `model_copy` instead. '
+            'See the docstring of `BaseModel.copy` for details about how to handle `include` and `exclude`.',
+            DeprecationWarning,
+        )
+
+        values = dict(
+            self._iter(to_dict=False, by_alias=False, include=include, exclude=exclude, exclude_unset=False),
+            **(update or {}),
+        )
+
+        # new `__fields_set__` can have unset optional fields with a set value in `update` kwarg
+        if update:
+            fields_set = self.__fields_set__ | update.keys()
+        else:
+            fields_set = set(self.__fields_set__)
+
+        # removing excluded fields from `__fields_set__`
+        if exclude:
+            fields_set -= set(exclude)
+
+        return self._copy_and_set_values(values, fields_set, deep=deep)
+
+    @classmethod
+    def schema(
+        cls, by_alias: bool = True, ref_template: str = DEFAULT_REF_TEMPLATE
+    ) -> typing.Dict[str, Any]:  # noqa UP006
+        warnings.warn('The `schema` method is deprecated; use `model_json_schema` instead.', DeprecationWarning)
+        return cls.model_json_schema(by_alias=by_alias, ref_template=ref_template)
+
+    @classmethod
+    def schema_json(
+        cls, *, by_alias: bool = True, ref_template: str = DEFAULT_REF_TEMPLATE, **dumps_kwargs: Any
+    ) -> str:
+        warnings.warn(
+            'The `schema_json` method is deprecated; use `model_json_schema` and json.dumps instead.',
+            DeprecationWarning,
+        )
+        from .json import pydantic_encoder
+
+        return cls.model_config['json_dumps'](
+            cls.model_json_schema(by_alias=by_alias, ref_template=ref_template),
+            default=pydantic_encoder,
+            **dumps_kwargs,
+        )
+
+    @classmethod
+    def validate(cls: type[Model], value: Any) -> Model:
+        warnings.warn('The `validate` method is deprecated; use `model_validate` instead.', DeprecationWarning)
+        return cls.model_validate(value)
+
+    @classmethod
+    def update_forward_refs(cls, **localns: Any) -> None:
+        warnings.warn(
+            'The `update_forward_refs` method is deprecated; use `model_rebuild` instead.', DeprecationWarning
+        )
+        cls.model_rebuild(force=True, types_namespace=localns)
+
+    # Deprecated: This private method is only used for `BaseModel.copy`, which is deprecated
+    def _iter(
```

Comment:
Can we move all this logic into the deprecated module as standalone functions?

Reply:
good idea, I'll do that.

Reply:
Moved these into deprecated/copy_internals.py, and retained the methods on the class with deprecation warnings. Let me know if you also want the methods themselves to have deprecation warnings. (If we do that, I think it may mean that calls to `Model.copy` trigger a bunch of warnings though)